### ISSUE TITLE
overlapping tags for Sequence Labeling

### ISSUE THREAD

**Issue by anbrun** [zalandoresearch/flair#902]
Hello!
I have been training my own Sequence Labeling model using SequenceTagger sucessfully with simple tags, essentially associating each token with one tag:
E.g. This_tag1 is_tag1 a_tag2 Test_tag1

Now I am wondering whether it would also be possible to use more complex structures as a learning target. Specifically, I would like to be able to train a model that can detect multiple overlapping tags:
E.g. This_[tag1] is_[tag1] a_[tag1, tag2] Test_[tag1].
Or maybe use layers for the tag1 (Position 1) and tag2 (Position 2) level:
E.g. This_[1, 0] is_[1, 0] a_[1, 1] Test_[1, 0].

Ist something like this possible?

**Comment by alanakbik** [comment]
Hello @anbrun we do not have integrated support for this currently. What you can do is train two independent sequence labelers (one for each layer) or try to 'merge' tags to create a new tag system that consists of tag combinations. How well these solutions would work would depend on your data and problem.

**Comment by alanakbik** [comment]
Great, will let you know - we'd also be interested to hear how it works with independent labelers!

**Comment by anbrun** [comment]
Independent labelers work decently, I guess. At the moment I train one for each category, always labelling the data with "tag - not-tag". The success rates vary by category. 
As the categories are related though and tend to occur in certain patterns (e.g. cat1 often contains cat2), I wanted to try learning them all at once and then realized that sequence labelling tasks seem to use 'flat' tags in general. I noticed that you provide a model trained on Propbank which seems to use somewhat complex tags (like <return.01>) but I guess that was learned as a single unit then and for the tagger there is no internal relation between, say,<return.01> and <return.02>.

**Comment by alanakbik** [comment]
Yes the Propbank model only does word-level prediction of one word per tag, so unfortunately no hierarchy there yet.
